update main README file (#75)
nickmoreton/wagtail-wordpress-import: README.md

    - [First steps to configure your wagtail app](#first-steps-to-configure-your-wagtail-app)
  - [Running the import command](#running-the-import-command)
    - [Optional command arguments](#optional-command-arguments)
  - [Module documentation](#module-documentation)
  - [Developer Tooling](#developer-tooling)
    - [Useful Shell Commands](#useful-shell-commands)
      - [Delete all imported images](#delete-all-imported-images)
      - [Delete all imported documents](#delete-all-imported-documents)

## Requirements

1. Wagtail CMS Installed with initial setup
2. Wordpress XML export of all content in a single file
3. The Wordpress website will need to be live and available for importing of assets such as images and documents.

## Initial app and package setup

1. Setup a Wagtail site using your preferred method or follow the [official documentation](https://docs.wagtail.io/en/stable/getting_started/tutorial.html) to get started.
2. Download the repo for this package [Wagtaill Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)

> 3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.

Reviewer: **Question**: Why can't this be installed with `pip install -e git+https://github.com/torchbox/wagtail-wordpress-import/`, i.e. without downloading?

**Suggestion**: Fix typo.

```suggestion
2. Download the repo for this package [Wagtail Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.
```

Author: > Why can't this be installed with `pip install -e git+https://github.com/torchbox/wagtail-wordpress-import/`, i.e. without downloading?

That makes sense but I see:
```
ERROR: Could not detect requirement name for 'git+https://github.com/torchbox/wagtail-wordpress-import/', please specify one with #egg=your_package_name
```

Reviewer: Ah, yes, sorry. For me locally it works with

```
pip install -e "git+https://github.com/torchbox/wagtail-wordpress-import.git#egg=wagtail-wordpress-import"
```

4. Place your XML files somewhere on your disk. The file can have any name you choose.

### First steps to configure your wagtail app

The import can be run on an existing or new site but you will need to perform some setup on your page models.

> We recommend your page model inherits from the provided WPImportedPageMixin

Reviewer: **Suggestion**: It must declare an `import_wordpress_data` method.

However, I'm not sure how I feel about the implementation, see my comment on that method below.

Author: New ticket created to alter this: https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/85

```python
from wagtail_wordpress_import.models import WPImportedPageMixin
class PostPage(WPImportedPageMixin, Page):
    ...
```

You will need to run `python manage.py makemigrations` and `python manage.py migrate` to add the fields to your page model.

> *It's intended that this initial setup can be removed once the content has been imported. [view source](wagtail_wordpress_import/models.py)*

Reviewer: ```suggestion
*It's intended that these fields are temporary for while importing, and can be removed once the content has been imported. [view source](wagtail_wordpress_import/models.py)*
```

    content_panels = Page.content_panels + [
        StreamFieldPanel("body"),
    ]

    edit_handler = TabbedInterface(
        [
            ObjectList(content_panels, heading="Content"),
            ObjectList(Page.promote_panels, heading="Promote"),
            ObjectList(Page.settings_panels, heading="Settings", classname="settings"),
            ObjectList(WPImportedPageMixin.wordpress_panels, heading="Debug"),
        ]
    )

    def import_wordpress_data(self, data):
        # wagtail page model fields
        self.title = data["title"]
        self.slug = data["slug"]
        self.first_published_at = data["first_published_at"]
        self.last_published_at = data["last_published_at"]
        self.latest_revision_created_at = data["latest_revision_created_at"]
        self.search_description = data["search_description"]

        # debug fields
        self.wp_post_id = data["wp_post_id"]
        self.wp_post_type = data["wp_post_type"]
        self.wp_link = data["wp_link"]
        self.wp_raw_content = data["wp_raw_content"]
        self.wp_block_json = data["wp_block_json"]
        self.wp_processed_content = data["wp_processed_content"]

>         self.wp_normalized_styles = data["wp_normalized_styles"]

Reviewer: **Comment**: It seem odd to expect the implementer to declare all these, when they're inheriting from a supplied base class. Can't we have the Wagtail Page model fields, and the debug fields, in a parent class method somewhere, and require the implementer to specify only the body field below?

Author: I have created a ticket to alter this: https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/85

        # own model fields
        self.body = data["body"]
```

## Running the import command

The most basic command would be:

```bash

> `python manage.py import_xml path/to/xml/file.xml parent_page_id`

Reviewer: **Suggestion**: Remove backticks inside code block.

```suggestion
python manage.py import_xml path/to/xml/file.xml parent_page_id
```

```

[parent_page_id] is the id of the page in your wagtail site where pages will be created as child pages. If you don't know what that is then choose to edit the page in wagtail admin and look at the url in your browser e.g. `http://www.domain.com/admin/pages/3/edit/` the number after /admin/pages/`3` is the id to use.

Running this command will import all Wordpress 'post' and 'page' types to the 'PostPage' model in and app called 'pages'

### Optional command arguments

- `-m` can be used to specify the Wagtail Page model to use for all imported pages. The default is `PostPage` when it's not specified.
- `-a` can be used to specify the Wagtail App where you have created your Wagtail Page model. The default is `pages` when it's not specified.
- `-t` can be used to limit the Wordpress page types to be imported. You can pass in a comma separated string of page types or just a single page type. The default is `page,post` if not specified.
- `-s` can be used to specify the status of pages you want to import. You can pass in a comma separated string of statuses or just a single status. The default is `publish,draft` if not specified.

> [View the import command source](wagtail_wordpress_import/management/commands/import_xml.py) to if you need to extend the command.

Reviewer: **Issue**: I'm not sure what this sentence is intended to say.

Author: I think one of my just in case suggestions again. Will remove it.

```bash
python manage.py reduce_xml path/to/your/xmlfile.xml
```

 will remove all items of `<wp:comment>`

---

To understand more about the type of HTML content formatting used in your Wordpress site we provide a reporting tool that will generate a table output in the console to show inline styles, html tags and shortcodes.

```bash
python manage.py analyze_html_content path/to/your/xmlfile.xml
```

---

When testing imports you may need to remove all the imported pages and run it again. This script will run until all pages have been deleted and display progress in the console.

```bash
python manage.py delete_imported_pages [app] [page_model]
# app and page_model are required arguments
```

### Useful Shell Commands

Start a shell with `python manage.py shell`

#### Delete all imported images

```shell
from wagtail.images.models import Image
+
Image.objects.all().delete()
```

#### Delete all imported documents

```shell
from wagtail.documents.models import Document
+
Document.objects.all().delete()

> ```

Reviewer: **Issue**: These commands both look like they will delete _all_ images and documents.

## Initial app and package setup

1. Setup a Wagtail site using your preferred method or follow the [official documentation](https://docs.wagtail.io/en/stable/getting_started/tutorial.html) to get started.
2. Download the repo for this package [Wagtail Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
3. Install this package with pip install -e "git+https://github.com/torchbox/wagtail-wordpress-import.git#egg=wagtail-wordpress-import"

>  or using any method you prefer.

Reviewer: **Issue**: Now we're telling people to install from GitHub, they won't need the download step.

Author: Of course :) I removed it.
